lib/modules/manager/pip-compile/extract.ts

```diff
@@ -200,6 +200,30 @@ export async function extractAllPackageFiles(
     depsBetweenFiles,
     packageFiles,
   );
+
+  const lockFileSources = new Map<string, PackageFile>();
+  for (const packageFile of result) {
+    for (const lockFile of packageFile.lockFiles!) {
+      lockFileSources.set(lockFile, packageFile);
+    }
+  }
```

Comment:
can't you fill the map in the loop above while extracting? This is even more bad to loop again. 🤔